### ISSUE TITLE
Ability to set stack creation timeout

### ISSUE THREAD

**Issue by greg5123334** [aws/aws-cdk#21606]
### Describe the feature

In pure Cloudformation, one can control the amount of time that can pass before the stack status becomes CREATE_FAILED by means of the [--timeout-in-minutes](https://docs.aws.amazon.com/cli/latest/reference/cloudformation/create-stack.html) CLI flag. This feature however, does not seem to be available in CDK.

### Use Case

Just as in Cloudformation, it would be nice to be able to define stack creation timeouts when working with the CDK.

### Proposed Solution

Add a property to the `App` class to allow one to define a stack creation timeout.

And/or

add a `--timeout-in-minutes` argument to the `cdk deploy` toolkit command.

### Other Information

Despite not being able to find documentation to support this, it seems possible to set stack creation timeouts in Python by doing the following:

```
app = cdk.App()
app.timeout = 1
```

And by this i mean the interpret doesn't complain about this pproerty and deployment can happen, however during stack creation itself the timeout  is **not** honoured.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.37.1 (build f15dee0)

### Environment details (OS name and version, etc.)

Windows 10

**Comment by peterwoodworth** [comment]
We use the `createChangeSet` operation on deploy - not the `CreateStack` operation. Since [`CreateChangeSet`](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_CreateChangeSet.html) doesn't have this option, I'm not sure there's an easy way for us to implement this feature without direct support on the operation we use.

**Comment by greg5123334** [comment]
thanks once again for the awesome help Peter. and that is really interesting to learn about using the createChangeSet command, today i learnt something.

i understand it would be a bit of a struggle and we may need the Cfn team's help to support this option on the CreateChangeSet command. 

feel free to close this ticket and re-open when appropriate.

**Comment by peterwoodworth** [comment]
This is a valid feature request, so I will leave it open 🙂 We can still track issues here that are blocked by CloudFormation

**Comment by greg5123334** [comment]
AWeSomeness..thanks Peter. and if you need anything you know how to reach me ;)

**Comment by houston3** [comment]
If possible, could you also add timeout and retry configuration to updates, not just creation? 
One of the most frustrating things about CDK and Cloudformation is waiting for a failed operation to complete and rollback.

**Comment by rgoltz** [comment]
It's a valid feature request (which needs implementation on CFN side first)

**Comment by aemada-aws** [comment]
@rgoltz I understand it is a valid feature request, but to me it looks like a duplicate of https://github.com/aws/aws-cdk/issues/24792 which has more engagement. Please correct me if i'm wrong.
